### PR TITLE
Add code to MR for tcp_max_syn_backlog test

### PR DESCRIPTION
**Pull request by nsednev**
Title: Validate if tcp_max_syn_backlog network parameter on Host can be resilient against random changes of this parameter in QM.

Description: The purpose of this test script is to validate if tcp_max_syn_backlog parameter value on Host, can be resilient against random changes of this parameter in QM. This is validated by creating a test environment, that generates and sets a random tcp_max_syn_backlog value in range from 128 to 1024 in QM and as soon as QM gets a new value, the test verifies that tcp_max_syn_backlog parameter on Host stays the same and not changes because of the QM network changes.

Expected result: PASS: tcp_max_syn_backlog parameter's value on Host stays the same.

Jira: VROOM-19564

### CODE REVIEW

File: tests/ffi/tcp_max_syn_backlog/test.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+. ../common/prepare.sh
+
+# Declare global variables and initialize to 0
```
Comment (dougsland) [inline tests/ffi/tcp_max_syn_backlog/test.sh:6]: Hi, nice patch.

Small comment. Probably these global variables need to be moved to main.fmf as env vars. :)

```
summary: Check that changes to network settings inside QM doesn't change network settings
    on Host.
test: /bin/bash ./test.sh
environment:
QM_NET_IPV4_TCP_MAX_SYN_BACKLOG_VALUE=0
HOST_DEFAULT_NET_IPV4_TCP_MAX_SYN_BACKLOG_VALUE=0
HOST_NET_IPV4_TCP_MAX_SYN_BACKLOG_VALUE=0
duration: 10m
tag: ffi
framework: shell
id: 7b86ccc3-072d-40b6-8463-3f24ceaba9b2
```

than remove it from the bash script and should be automatically available in the bash script, please test it to make sure it works.

@Yarboa could you please review if you have time before flying?

**Comment by dougsland** [comment]
looks like it reached the timeout limit (15m), we need to increase it.
```
Maximum test time '15m' exceeded.
Adjust the test 'duration' attribute if necessary.
https://tmt.readthedocs.io/en/stable/spec/tests.html#duration
```

File: tests/ffi/tcp_max_syn_backlog/test.sh (hunk 1 above)
Comment (nsednev) [reply tests/ffi/tcp_max_syn_backlog/test.sh:6]: Fixed and changed README.md.

**Comment by nsednev** [comment]
> looks like it reached the timeout limit (15m), we need to increase it.
> 
> ```
> Maximum test time '15m' exceeded.
> Adjust the test 'duration' attribute if necessary.
> https://tmt.readthedocs.io/en/stable/spec/tests.html#duration
> ```

Changed to 20 mins and works fine now.
I'm not sure why that happened, might be the labe took more time Yesterday for some reason.

File: tests/ffi/tcp_max_syn_backlog/test.sh (hunk 1 above)
Comment (nsednev) [reply tests/ffi/tcp_max_syn_backlog/test.sh:6]: Moving global variables to main.fmf causing for test failure on testing-farm:
Failed to discover tests, TMT metadata are absent or corrupted. Search for '[E] [test-schedule-tmt' on the artifacts page for details.
If possible, I'd prefer to leave them inside the test.sh.

I've tried with this syntax in main.fmf and commented out these variables in test.sh:
summary: Check that changes to network settings inside QM doesn't change network settings
    on Host.
test: /bin/bash ./test.sh
environment:
QM_NET_IPV4_TCP_MAX_SYN_BACKLOG_VALUE=0
HOST_DEFAULT_NET_IPV4_TCP_MAX_SYN_BACKLOG_VALUE=0
HOST_NET_IPV4_TCP_MAX_SYN_BACKLOG_VALUE=0
duration: 20m
tag: ffi
framework: shell
id: 7b86ccc3-072d-40b6-8463-3f24ceaba9b2

File: tests/ffi/tcp_max_syn_backlog/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Title:
+    Validate if tcp_max_syn_backlog network parameter on Host can be resilient against random changes of this parameter in QM.
+
+Description:
+    The purpose of this test script is to validate if tcp_max_syn_backlog parameter value on Host, can be resilient against random changes of this parameter in QM. This is validated by creating a test environment, that generates and sets a random tcp_max_syn_backlog value in range from 128 to 1024 in QM and as soon as QM gets a new value, the test verifies that tcp_max_syn_backlog parameter on Host stays the same and not changes because of the QM network changes.
+
+Input:
+    Environment variables:
```
Comment (dougsland) [inline tests/ffi/tcp_max_syn_backlog/README.md:8]: If you are not using env variables, I think just set this as blank:

Input:

File: tests/ffi/tcp_max_syn_backlog/main.fmf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+summary: Check that changes to network settings inside QM doesn't change network settings
+    on Host.
+test: /bin/bash ./test.sh
+duration: 20m
```
Comment (dougsland) [inline tests/ffi/tcp_max_syn_backlog/main.fmf:4]: I still see the error, I am not sure why I still see the error in the github and not respecting this 20m

**Comment by dougsland** [comment]
It's Friday and almost end of the day for some countries in Europe, @dennisbrendel might have some ideas if he is around. Otherwise, I will try to look dig it later today. I think Yariv is already kind of busy travelling outside his country for the conference to review this one.

File: tests/ffi/tcp_max_syn_backlog/main.fmf (hunk 3 above)
Comment (nsednev) [reply tests/ffi/tcp_max_syn_backlog/main.fmf:4]: And the weirdest thing is that I had it passed just fine even without increasing the time.
Its also wrote in the attributes that "/tests/ffi/memory (timeout)took 900 s"
My test has nothing related to memory, its /ffi/tcp_max_syn_backlog.
Yesterday I saw that it worked fine even with 10 min test run.
Screenshot from 2024-08-08 20-57-55
I've also tested it on TC and it works.
Lets see if it works now...

**Comment by nsednev** [comment]
> It's Friday and almost end of the day for some countries in Europe, @dennisbrendel or @priyverm might have some ideas if he is around. Otherwise, I will try to look dig it later today. I think Yariv is already kind of busy travelling outside his country for the conference to review this one.

Yes, its a holiday in here Today, I'm just trying to cover what I can till the next week.
Really no need to hurry, we can push it next week.

File: tests/ffi/tcp_max_syn_backlog/README.md (hunk 2 above)
Comment (nsednev) [reply tests/ffi/tcp_max_syn_backlog/README.md]: Cleared up and removed.

File: tests/ffi/tcp_max_syn_backlog/main.fmf (hunk 3 above)
Comment (nsednev) [reply tests/ffi/tcp_max_syn_backlog/main.fmf:4]: I see that it works now.
Looks like some infra issues with unrelated to the test other test runs.

Comment (dougsland) [inline tests/ffi/tcp_max_syn_backlog/main.fmf:4]: Could you please send another patch to increase to 30m ?

**Review by dougsland — APPROVED** [review]
